### DISCUSSION TITLE
Mods

### DISCUSSION THREAD

**Discussion by alexanderson1993** [Thorium-Sim/thorium-nova#32] [Ideas] [answered]
This is a slightly different conversation than https://github.com/Thorium-Sim/thorium-nova/discussions/26, which are intended to be content packs. Mods, on the other hand, would add code to fundamentally change how the game works. 

It's a simple discussion, really. Should Mods be a thing, or are content packs good enough?

Pros:
- Ability to add new features without changing the core.
- Ability to share those changes with others.

Cons:
- Complicated to implement, especially for the UI (suggestions for how to implement this are welcome)
- Possibly redundant, since the controls are open source anyway - mods should just be implemented in the core.
  - This might require a feature flag setup to allow people to turn on and off certain features (although I think that's a good idea anyway)

Any thoughts? The default would be no mods, but I want to hear if anyone has compelling arguments and suggestions for or against.

**Comment by astrolamb-gaming** [comment]
I had a whole argument for including mods, but honestly you're right - if a feature is desirable enough to make a mod for it, it should go into the core game anyhow. 

I do think there should be a way for players (or a flight director) to select which features they want included in a particular game. 

A subsequent logical feature would be a "base" set of features for beginners to use to learn the basics of controls and teamwork in the game before being swamped with tons of features

**Reply by alexanderson1993** [reply]
> I had a whole argument for including mods...

In my mind, the whole argument is "monolith" vs "lean core". Benefits and drawbacks to both, but I think Monolith is easier to execute, and plugins will provide more than enough opportunity for customization outside of editing the code itself.

I want to give it more time for comments from others, but I think a feature flag system should be added to the project. I'll write up an issue in the next day or so.

**Comment by alexanderson1993** [comment, marked as answer]
Marking this as answered. Mods are a no-go. Plugins are the only way to customize, other than contributing to the core. 

I'll add an issue for a feature flag system.
